jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/rtree/InnerNode.java

```diff
@@ -135,8 +134,9 @@ public Rectangle2D getBounds() {
   @Override
   public Node<T> recalculateBounds() {
     bounds = Optional.empty();
-    for (Node<T> kid : children) {
-      updateBounds(kid.getBounds());
+    int size = children.size();
```

Comment:
I see the performance data that you quoted above (radius vs. spatial), but I don't see how that relates to the foreach changes made in this commit, i.e., they're not making the case that moving away from the foreach loops actually helped.

I'm not going to quibble about these changes, but I generally prefer the foreach format for readability and conciseness.  (And I'm willing to trust that the JVM will continue to optimize these things inline anyway.)

Reply:
I did not seek to prove anything by just making a sensible (to me) change in the code.
I do some research about game programming and other domains where performance is critical. At the beginning, I was horrified to see that my RTree was actually slower. It was because I was creating too many objects (collections in that case). For critical code (anything in an algorithm) I will remove code that creates objects that are not necessary (like Iterators in this case).
It comes down to this:
Creating objects costs > 0 time
Not creating superfluous objects in critical code will always be faster.
I don't have to prove that. You have to prove the converse, if readability and cleaner language features are so important to you that you will hope for the best with performance.

Reply:
I assumed that the performance data that you quoted were trying to make the argument for this change; otherwise I couldn't see why you'd included them in the commit message. 

The JVM is perfectly capable, in theory at least, of recognizing that you've got a List and converting that code into something that iterates over it without using an Iterator.  I don't know that it does so, but it's an easy optimization on their side and it wouldn't surprise me.

Again, I'm not asking you to revert these changes, but I don't think that it's unreasonable to have asked for performance data...since it appeared that you had a mechanism already for calculating performance stats.

Reply:
other changes were responsible for the major performance improvement. I just went after every tweak I could find and for-loops were just one of them. I was happy with the timing measurement, so i documented it in the commit message.
I did know that removing for-each would be noticed and questioned.